source/AMDSnDIndex.h

```diff
@@ -256,14 +256,14 @@ bool AMDSnDIndex::operator!=(const AMDSnDIndex& other) const {
 
 quint32 AMDSnDIndex::product() const {
 	quint32 rv = 1;
-	for(quint8 mu=rank_-1; mu>=0; --mu)
+	for(qint8 mu=rank_-1; mu>=0; --mu)
```

Comment:
There is a compiling warning for this.

X.L.

From: Darren Hunter [mailto:notifications@github.com]
Sent: July-13-15 1:03 PM
To: acquaman/AcquamanDataServer
Cc: Xuetao (Sheldon) Liu
Subject: Re: [AcquamanDataServer] Pull Request for Issue2 (#36)


In source/AMDSnDIndex.h<https://github.com/acquaman/AcquamanDataServer/pull/36#discussion_r34498968>:

> @@ -256,14 +256,14 @@ bool AMDSnDIndex::operator!=(const AMDSnDIndex& other) const {

>

>  quint32 AMDSnDIndex::product() const {

>       quint32 rv = 1;

> -     for(quint8 mu=rank_-1; mu>=0; --mu)

> +     for(qint8 mu=rank_-1; mu>=0; --mu)

I know this is already merged in, but why was this changed to a qint8 rather than changing rank_ to a quint8? Rank should never be negative.

—
Reply to this email directly or view it on GitHub<https://github.com/acquaman/AcquamanDataServer/pull/36/files#r34498968>.